packages/sdks/react-sdk/src/components/WebComponentBridge.tsx

```diff
@@ -0,0 +1,78 @@
+import React, {
+  useImperativeHandle,
+  useMemo,
+  ComponentType,
+  useCallback,
+  useRef,
+} from 'react';
+import { kebabCase } from '@descope/sdk-helpers';
+
+const propSuffixes = {
+  prop: '.prop',
+  attr: '.attr',
+};
+
+/**
+ * WebComponentBridge is a React HOC that adapts React props to work seamlessly
+ * with web components by setting attributes and properties.
+ *
+ * - Props ending in `.prop` are set as properties on the web component.
+ * - Props ending in `.attr` are set as attributes on the web component (kebab-case).
+ * - All other props are set as props
+ *
+ * This resolves attribute/property behavior differences in React 19.
+ *
+ * @see https://github.com/facebook/react/issues/29037
+ */
+const WebComponentBridge = <P extends Record<string, any>>(
+  Component: ComponentType<any>,
+) =>
+  React.forwardRef<HTMLElement, P>((props, ref) => {
+    const compProps = useMemo(
+      () =>
+        Object.entries(props || {}).reduce((acc, [key, value]) => {
+          if (
+            key.endsWith(propSuffixes.attr) ||
+            key.endsWith(propSuffixes.prop)
+          ) {
+            return acc;
+          }
+          return { ...acc, [kebabCase(key)]: value };
+        }, {}),
+      [props],
+    );
```

Comment:
What about transforming the props before, so the CB fn will be much simple? 
```suggestion
  const compProps = useMemo(() => {
  // can be extracted into helper file
    const transformKey = (key) => {
      const [, trimmedKey, category] = /(.*)\.(prop|attr)$/.exec(key) || [
        ,
        key,
        'rest',
      ];

      return { trimmedKey, category };
    };
    
  // can be extracted into helper file
    const transformAttrValue = (value) =>
      typeof value === 'string' || value == null
        ? value
        : JSON.stringify(value);

    return Object.entries(props).reduce(
      (acc, [key, value]) => {
        const { trimmedKey, category } = transformKey(key);
        if (category === 'prop') acc.prop.push([trimmedKey, value]);
        else if (category === 'attr')
          acc.attr.push([kebabCase(trimmedKey), transformAttrValue(value)]);
        else Object.assign(acc.rest, { [trimmedKey]: value })

        return acc;
      },
      { attr: [], prop: [], rest: {} },
    );
  }, [props]);

```

Reply:
thanks!

Reply:
great suggestions, thanks for that!